### PR TITLE
Updated AngularJs orderBy definition to version 1.5.8

### PR DESCRIPTION
**Pull request by Venthe**
case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url http://api.jquery.com/html .
  - it has been reviewed by a DefinitelyTyped member.

### CODE REVIEW

**Comment by thorn0** [comment]
This type of the added `comparator` parameter isn't `(v1: T, v2: T) => number`. Please read the documentation https://docs.angularjs.org/api/ng/filter/orderBy
Also please use spaces for indentation, don't mix them with tabs.

**Comment by Venthe** [comment]
I blame N++ for tabs :)

Done.

**Comment by thorn0** [comment]
The type of the `comparator` parameter is still wrong.

**Comment by Venthe** [comment]
@thorn0 could you clarify? I'm not used to write in either JS or TS. As far as I can tell, it is OK now, or it was even better before. It does not help if you tell me "You've done it wrong, do it again" because I will not do it again, and this typing will be out of date as it was before. I've tried to fix it one more time.

File: angularjs/angular.d.ts [hunk 1]
```diff
@@ -648,13 +648,13 @@ declare namespace angular {
     }
 
     interface IFilterFilterPatternObject {
-        value: any;
```
Comment (Venthe) [inline angularjs/angular.d.ts]: It's a revert

**Comment by thorn0** [comment]
Filters, in particular this one, are used almost exclusively in templates, which unfortunately aren't type checked at all. Do you have a real-life use case for these type definitions?

**Comment by Venthe** [comment]
Yes I do – I had to use It at my work to override string comparator for grid, to include accented letters.

 

To achieve that, I had to skip angular.d.ts interface for angular’s filter entirely and set it to any – which is obviously a no go.

 

I’ve used comparator like this

Comp(val:any, val2:any) : number {

val=val.value; // Object really is {value, type, id}

val2=val2.value;

 

//Code for intl comparator which returns -1,0,1

}

 

And I really don’t have much more experience in TS, so even if I want to write a good code here, I’m kind of stumbling around in unknown regions for me :)

File: angularjs/angular.d.ts [hunk 2]
```diff
@@ -651,6 +651,12 @@ declare namespace angular {
         [name: string]: any;
     }
 
+    interface IFilterOrderByObject {
+        value: any;
+        type: any;
+        index: any;
+    }
```
Comment (aciccarello) [inline angularjs/angular.d.ts]: Shouldn't these have the types `any`, `string`, and `number` respectively?

File: angularjs/angular.d.ts [hunk 3]
```diff
@@ -659,6 +665,10 @@ declare namespace angular {
         (actual: T, expected: T): boolean;
     }
 
+    interface IFilterOrderByComparatorFunc {
+        (value1: IFilterOrderByObject, value2: IFilterOrderByObject): number;
```
Comment (aciccarello) [inline angularjs/angular.d.ts]: It might be helpful to document the expected return numbers in a jsdoc comment.

**Review by aciccarello — CHANGES_REQUESTED** [review]
Looks like it's coming along. Also should add test with simple compare function.

File: angularjs/angular.d.ts (hunk 2 above)
Comment (thorn0) [inline angularjs/angular.d.ts]: `value` is a return value of one of the predicates (including the 'dummy predicate', see the docs). Different predicates can return values of different types. Also predicates can be expressions (strings), not functions. So I don't think it's possible to think of something better than `any` here.

Comment (thorn0) [inline angularjs/angular.d.ts]: We could probably type the `predicates` parameter as `(string | ((value: T) => U))[] | (value: T) => U | string` if type arguments could be inferred to be union types in the cases like `[x => 1, x => 'a']`. But unfortunately the inference [doesn't work like this](http://stackoverflow.com/questions/39905523/why-isnt-the-type-argument-inferred-as-a-union-type).

**Comment by Venthe** [comment]
I do not know how to write tests like used here, so if anyone would be willing to help me, I'm open.

**Comment by minestarks** [comment]
@Venthe simple -- add some sample code that uses your type definition changes to angular-tests.ts file in this directory. If the code compiles the tests pass.

**Comment by paulvanbrenk** [comment]
This PR seems abandoned, please re-opeon when you made the requested changes.

**Pull request closed without merging**